jetty-websocket/jetty-websocket-server/src/main/java/org/eclipse/jetty/websocket/server/JettyServerFrameHandlerFactory.java

```diff
@@ -18,20 +18,44 @@
 
 package org.eclipse.jetty.websocket.server;
 
+import org.eclipse.jetty.server.handler.ContextHandler;
+import org.eclipse.jetty.util.component.LifeCycle;
 import org.eclipse.jetty.websocket.common.JettyWebSocketFrameHandlerFactory;
 import org.eclipse.jetty.websocket.core.FrameHandler;
 import org.eclipse.jetty.websocket.server.internal.UpgradeRequestAdapter;
 import org.eclipse.jetty.websocket.server.internal.UpgradeResponseAdapter;
+import org.eclipse.jetty.websocket.servlet.FrameHandlerFactory;
 import org.eclipse.jetty.websocket.servlet.ServletUpgradeRequest;
 import org.eclipse.jetty.websocket.servlet.ServletUpgradeResponse;
-import org.eclipse.jetty.websocket.servlet.WebSocketServletFrameHandlerFactory;
 
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.Executor;
 
-public class JettyWebSocketServletFrameHandlerFactory extends JettyWebSocketFrameHandlerFactory implements WebSocketServletFrameHandlerFactory
+import javax.servlet.ServletContext;
+
+public class JettyServerFrameHandlerFactory
+    extends JettyWebSocketFrameHandlerFactory
+    implements FrameHandlerFactory, LifeCycle.Listener
 {
-    public JettyWebSocketServletFrameHandlerFactory(Executor executor)
+    public static JettyServerFrameHandlerFactory ensureFactory(ServletContext servletContext)
+    {
+        ContextHandler contextHandler = ContextHandler.getContextHandler(servletContext);
```

Comment:
`ContextHandler.getContextHandler(servletContext)` doesn't work reliably on mock libraries, spring, spark, etc...

You need a safety net, throwables when these environments are used.

``` java
if (contextHandler == null)	
{	
    throw new ServletException("Not running on Jetty, Javax Websocket support unavailable");	
}	

if (!(contextHandler instanceof ServletContextHandler))	
{	
    throw new ServletException("Not running in Jetty ServletContextHandler, Javax Websocket support unavailable");	
}	
```